cmd/podman/kube/down.go

```diff
@@ -0,0 +1,61 @@
+package pods
+
+import (
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/containers/podman/v4/cmd/podman/common"
+	"github.com/containers/podman/v4/cmd/podman/registry"
+	"github.com/spf13/cobra"
+)
+
+var (
+	downDescription = `Reads in a structured file of Kubernetes YAML.
+
+  Removes pods that have been based on the Kubernetes kind described in the YAML.`
+
+	downCmd = &cobra.Command{
+		Use:               "down KUBEFILE|-",
+		Short:             "Remove pods based on Kubernetes YAML.",
+		Long:              downDescription,
+		RunE:              down,
+		Args:              cobra.ExactArgs(1),
+		ValidArgsFunction: common.AutocompleteDefaultOneArg,
+		Example: `podman kube down nginx.yml
+   cat nginx.yml | podman kube down -`,
+	}
+)
+
+func init() {
+	registry.Commands = append(registry.Commands, registry.CliCommand{
+		Command: downCmd,
+		Parent:  kubeCmd,
+	})
+}
+
+func down(cmd *cobra.Command, args []string) error {
+	tmpfile, err := ioutil.TempFile("", "stdin")
```

Comment:
The "stdin dance" should be conditional.
```go
if args[0] == "-" {
   // stdin
} else {
   // file
}
```

It's probably worth putting that into a dedicated function to avoid the duplicate code:
```golang
func getFileFromArg(arg string) (*os.File, bool, error)
```
The first return value is the file, the second `remove` indicates whether we should remove the file, and the last is the go-idiomatic error.

Reply:
Why do we need a tmp file at all? If we have to read it more than once we can just copy it into a bytes slice and then create a reader with bytes.NewReader().
I do not like tmpfiles since they stick around if the program is stopped/crashes before we remove it.

Reply:
Nice idea.  That'll work as well as it allows for seeking https://pkg.go.dev/bytes#Reader.Seek.